NetworkingServiceKit/Classes/Networking/NetworkManager.swift

```diff
@@ -9,131 +9,184 @@
     /// - internalServerError: a 500 request
     /// - badStatusCode: a request that could not validate a status code
-    /// - persistenceFailure: any issues related to our data layer
     /// - notConnected: user not connected to the internet
+    /// - forbidden: received a 403
+    /// - notFound: 404, path not found
+    /// - invalidResponse: server returns a response, but not in the expected format
     public enum ResponseFailureReason {
-
-        case tokenExpired(code: Int)
-        case badRequest(error: Error)
+        case tokenExpired
+        case badRequest
         case internalServerError
         case badStatusCode
-        case persistenceFailure(code: Int)
-        case connectivity(code: Int)
+        case connectivity
+        case forbidden
+        case notFound
+        case invalidResponse
+        
+        public var description: String {
+            switch self {
+            case .tokenExpired:
+                return "Expired Auth Token"
+            case .badRequest:
+                return "Invalid Network Request"
+            case .forbidden:
+                return "Invalid Credentials"
+            case .notFound:
+                return "Invalid Request Path"
+            case .internalServerError:
+                return "Internal Server Error"
+            case .badStatusCode:
+                return "Invalid Status Code"
+            case .connectivity:
+                return "Network Connectivity Issues"
+            case .invalidResponse:
+                return "Unable to Parse Server Response"
+            }
+        }
     }
-
+    
+    /// Enum used for errors associated with local failures
+    ///
+    /// - invalidData: Catchall for expected data being missing
+    /// - persistenceFailure: Core Data failure
     public enum PersistenceFailureReason {
-
         case invalidData
-        case persistenceFailure(code: Int)
+        case persistenceFailure
+        
+        public var description: String {
+            switch self {
+            case .invalidData:
+                return "Invalid Data"
+            case .persistenceFailure:
+                return "Core Data Failure"
+            }
+        }
     }
-
-    case responseValidationFailed(reason: ResponseFailureReason)
-    case persistenceValidationFailed(reason: PersistenceFailureReason)
-
+    
+    case responseValidation(reason: ResponseFailureReason)
+    case persistenceValidation(reason: PersistenceFailureReason)
 }
 
-// Mapped Error response failures
-extension MSError.ResponseFailureReason {
-    var responseCode: Int {
-        switch self {
-        case .tokenExpired(let code), .connectivity(let code):
-            return code
-        case .badRequest(let error):
-            return (error as NSError).code
-        case .internalServerError:
-            return 500
-        default:
-            return 0
-        }
+//Custom Makespace Error
+public struct MSError: Error {
```

Comment:
We might want to add some unit tests in objc as well, since I have found the library it's not fully compliant with objc. Specially writing unit tests in objc with stubbing it's not supported. But that's probably another task. What I would recommend anyhow is to test an api call from an objc source code and see that everything is reliable and accessible.